src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -234,7 +235,21 @@ void ConsoleProcess::commonInit()
                            options_.smartTerminal ? core::system::kSmartTerm :
                                                     core::system::kDumbTerm);
 
+      // environment variables that can be used when configuring custom shells
       core::system::setenv(&(options_.environment.get()), "RSTUDIO_TERM", procInfo_->getHandle());
+
+      core::system::setenv(&(options_.environment.get()), "RSTUDIO_PROJ_NAME",
+                           projects::projectContext().file().stem());
+
+      FilePath projPath = core::system::userHomePath();
```

Comment:
Just wanted a safe fallback, but I should make it consistent and use the same configured location.

Was thinking of use case of per-project-specific shell config file, potentially including one for "no project" case. To make this robust in multi-user/multi-session, guess they'd need to string together all these env variables (proj_path / $user-$session-myrc)?